src/ui/layout/tutorial.md

```diff
@@ -244,219 +393,287 @@ class MyApp extends StatelessWidget {
 
-<?code-excerpt "lib/main.dart (textSection)" title?>
+<?code-excerpt "step4/lib/main.dart (TextSection)" title?>
 ```dart
-Widget textSection = Container(
-  padding: const EdgeInsets.all(32),
-  child: const Text(
-    'Lake Oeschinen lies at the foot of the Blüemlisalp in the Bernese '
-    'Alps. Situated 1,578 meters above sea level, it is one of the '
-    'larger Alpine Lakes. A gondola ride from Kandersteg, followed by a '
-    'half-hour walk through pastures and pine forest, leads you to the '
-    'lake, which warms to 20 degrees Celsius in the summer. Activities '
-    'enjoyed here include rowing, and riding the summer toboggan run.',
-    softWrap: true,
-  ),
-);
+class TextSection extends StatelessWidget {
+  const TextSection({
+    super.key,
+    required this.description,
+  });
+
+  final String description;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(32),
+      child: Text(
+        description,
+        softWrap: true,
+      ),
+    );
+  }
+}
 ```
 
-By setting `softwrap` to true, text lines will fill the column width before
+By setting `softwrap` to `true`, text lines fill the column width before
 wrapping at a word boundary.
 
-Add the text section to the body:
+### Update the app to display the text section
 
-<?code-excerpt path-base="layout/lakes"?>
-<?code-excerpt "{step3,step4}/lib/main.dart" from="return MaterialApp"?>
+Insert the `TextSection` call into the `body` property.
+This call needs `description` variable set to the text of the
+location description.
+
+<?code-excerpt "step{3,4}/lib/main.dart (addWidget)" title?>
 ```diff
---- step3/lib/main.dart
-+++ step4/lib/main.dart
-@@ -59,3 +72,3 @@
-     return MaterialApp(
-       title: 'Flutter layout demo',
-       home: Scaffold(
-@@ -66,6 +79,7 @@
-           children: [
-             titleSection,
-             buttonSection,
-+            textSection,
-           ],
-         ),
+--- step3/lib/main.dart (addWidget)
++++ step4/lib/main.dart (addWidget)
+@@ -6,6 +6,16 @@
+         location: 'Kandersteg, Switzerland',
        ),
+       ButtonSection(),
++      TextSection(
++        description:
++            'Lake Oeschinen lies at the foot of the Blüemlisalp in the '
++            'Bernese Alps. Situated 1,578 meters above sea level, it '
++            'is one of the larger Alpine Lakes. A gondola ride from '
++            'Kandersteg, followed by a half-hour walk through pastures '
++            'and pine forest, leads you to the lake, which warms to 20 '
++            'degrees Celsius in the summer. Activities enjoyed here '
++            'include rowing, and riding the summer toboggan run.',
++      ),
+     ],
+   ),
+ ),
 ```
 
-## Step 5: Implement the image section
+## Add the Image section
 
-Three of the four column elements are now complete,
-leaving only the image. Add the image file to the example:
+In this section, add the image file to complete your layout.
```

Comment:
Similar to what I mentioned above, I think it would be helpful to be more clear in stating that the following sections will help them do that.

Reply:
It's more accurate to say "In this section" because it's the first line describing what the section does.

Reply:
Something like "In the following steps" or just "Next, you'll add" would also work.